### PR TITLE
fix bug with folders cleanup not being performed correctly

### PR DESCRIPTION
**Pull request by freedomp**
fix bug with folders cleanup not being performed correctly
on project import and upload zip project api

Signed-off-by: Dror Cohen dror.cohen@sap.com

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@vparfonov ?

**Comment by vparfonov** [comment]
ok

**Comment by skabashnyuk** [comment]
ok. Is it possible to add some tests?

**Comment by skabashnyuk** [comment]
@tareqhs can you merge with changes?

**Comment by tareksha** [comment]
@freedomp there are too many whitespace changes, please re-do the patch with minimal whitespace changes to make it clearer and easier to track

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 1]
```diff
@@ -805,15 +805,37 @@ public ImportResponse importProject(@ApiParam(value = "Workspace ID", required =
...
+        try {
+            baseProjectFolder = (FolderEntry)virtualFile;
+            importer.importSources(baseProjectFolder, projectSource.getLocation(), projectSource.getParameters(), outputOutputConsumerFactory);
+
+            //project source already imported going to configure project
+            return configureProject(importProject, baseProjectFolder, workspace, creationDate);
+        } catch (Exception e) {
+            // if import failed for some reason and the folder was created for this import
+            // remove it
+            if(!folderExistedBeforeCommand && baseProjectFolder != null) {
```
Comment (benoitf) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:822]: missing space after if( --> if (

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 2]
```diff
@@ -805,15 +805,37 @@ public ImportResponse importProject(@ApiParam(value = "Workspace ID", required =
...
+            throw e;
+        }
 
-        //project source already imported going to configure project
-        return configureProject(importProject, baseProjectFolder, workspace, creationDate);
     }
 
+    private boolean virtualEntryExists(String workspace, String path) throws ForbiddenException, ServerException, NotFoundException {
+        VirtualFileEntry virtualFile = projectManager.getProjectsRoot(workspace).getChild(path);
+        return (virtualFile != null);
```
Comment (benoitf) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:836]: we can use only one line
return projectManager.getProjectsRoot(workspace).getChild(path) != null

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 3]
```diff
@@ -982,23 +1003,36 @@ public ImportResponse uploadProjectFromZip(@ApiParam(value = "Workspace ID", req
...
-                                  .withRunners(new HashMap<String, RunnerSource>());
-        ImportProject importProject = dtoFactory.createDto(ImportProject.class)
-                                                .withProject(newProject)
-                                                .withSource(source);
-
-        //project source already imported going to configure project
-        return configureProject(importProject, baseProjectFolder, workspace, creationDate);
+        
+            final DtoFactory dtoFactory = DtoFactory.getInstance();
+            NewProject newProject = dtoFactory.createDto(NewProject.class)
```
Comment (benoitf) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:1012]: we've newDto(NewProject.class)  (newDto static import of DtoFactory class)

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 4]
```diff
@@ -982,23 +1003,36 @@ public ImportResponse uploadProjectFromZip(@ApiParam(value = "Workspace ID", req
...
+                                              .withName(projectName)
+                                              .withDescription(projectDescription)
+                                              .withVisibility(privacy);
+            Source source = dtoFactory.createDto(Source.class)
+                                      .withRunners(new HashMap<String, RunnerSource>());
```
Comment (benoitf) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:1017]: if it's to provide an empty map we may replace by Collections.emptyMap()

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 5]
```diff
@@ -982,23 +1003,36 @@ public ImportResponse uploadProjectFromZip(@ApiParam(value = "Workspace ID", req
...
+            ImportProject importProject = dtoFactory.createDto(ImportProject.class)
+                                                    .withProject(newProject)
+                                                    .withSource(source);
+
+            //project source already imported going to configure project
+            return configureProject(importProject, baseProjectFolder, workspace, creationDate);
+        } catch(Exception e) {
+            // if call failed for some reason and the folder was created for this action
+            // remove it
+            if(!folderExistedBeforeCommand) {
```
Comment (benoitf) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:1027]: missing space before (     "if ("

**Comment by tareksha** [comment]
hi @benoitf , i'll upload a new version of this change ASAP

**Pull request closed without merging**